presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergNativeMetadata.java

```diff
@@ -0,0 +1,523 @@
+    {
+        TableIdentifier from = toIcebergTableIdentifier(((IcebergTableHandle) tableHandle).getSchemaTableName());
+        TableIdentifier to = toIcebergTableIdentifier(newTable);
+        resourceFactory.getCatalog(session).renameTable(from, to);
+    }
+
+    @Override
+    public void addColumn(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnMetadata column)
+    {
+        TableIdentifier tableIdentifier = toIcebergTableIdentifier(((IcebergTableHandle) tableHandle).getSchemaTableName());
+        Table icebergTable = resourceFactory.getCatalog(session).loadTable(tableIdentifier);
+        icebergTable.updateSchema().addColumn(column.getName(), toIcebergType(column.getType())).commit();
+    }
+
+    @Override
+    public void dropColumn(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnHandle column)
+    {
+        TableIdentifier tableIdentifier = toIcebergTableIdentifier(((IcebergTableHandle) tableHandle).getSchemaTableName());
+        Table icebergTable = resourceFactory.getCatalog(session).loadTable(tableIdentifier);
+        IcebergColumnHandle handle = (IcebergColumnHandle) column;
+        icebergTable.updateSchema().deleteColumn(handle.getName()).commit();
+    }
+
+    @Override
+    public void renameColumn(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnHandle source, String target)
+    {
+        TableIdentifier tableIdentifier = toIcebergTableIdentifier(((IcebergTableHandle) tableHandle).getSchemaTableName());
+        Table icebergTable = resourceFactory.getCatalog(session).loadTable(tableIdentifier);
+        IcebergColumnHandle columnHandle = (IcebergColumnHandle) source;
+        icebergTable.updateSchema().renameColumn(columnHandle.getName(), target).commit();
+    }
+
+    private ConnectorTableMetadata getTableMetadata(ConnectorSession session, SchemaTableName table)
+    {
+        Table icebergTable;
+        try {
+            icebergTable = resourceFactory.getCatalog(session).loadTable(toIcebergTableIdentifier(table));
+        }
+        catch (NoSuchTableException e) {
+            throw new TableNotFoundException(table);
+        }
+
+        List<ColumnMetadata> columns = getColumnMetadatas(icebergTable);
+
+        ImmutableMap.Builder<String, Object> properties = ImmutableMap.builder();
+        properties.put(FILE_FORMAT_PROPERTY, getFileFormat(icebergTable));
+        if (!icebergTable.spec().fields().isEmpty()) {
+            properties.put(PARTITIONING_PROPERTY, toPartitionFields(icebergTable.spec()));
+        }
+
+        return new ConnectorTableMetadata(table, columns, properties.build(), getTableComment(icebergTable));
+    }
+
+    private List<ColumnMetadata> getColumnMetadatas(org.apache.iceberg.Table table)
+    {
+        return table.schema().columns().stream()
+                .map(column -> {
+                    return new ColumnMetadata(column.name(), toPrestoType(column.type(), typeManager), column.doc(), false);
```

Comment:
"return" is not necessary I think.  can we just write ".map(column -> new ColumnMetadata(column.name(), toPrestoType(column.type(), typeManager), column.doc(), false))"

Reply:
Similar to the last one, this is to match the behavior of `IcebergMetadata`, but that constructor is deprecated anyway, I will switch to use the builder.